### ISSUE TITLE
Cant run on debian

### ISSUE THREAD

**Issue by gadc1996** [the-lay/ulauncher-spotify-api#16]
After i install the extension via Ulauncher, i get the option to use it when `sp` is typed, the problem is that it never goes beyond the loading message, not even for the `?` command, 

Note: Autentication is never asked.

Any thought on what i am doing wrong?

**Comment by the-lay** [comment]
Sorry for the inconvenience! So if I understand correctly, the extension is successfully installed, but when you type `sp`, it gets stuck and nothing shows up.

Could you please open ulauncher's settings -> extension -> Ulauncher Spotify API and see if there are any warnings there?
Another thing you can try is running ulauncher verbosely in a terminal (`ulauncher -v`) and checking the output there when you type `sp`.

**Comment by the-lay** [comment]
> There are no warnings in the extension settings
> Do you think that this problem might be from ulauncher itself, because even when i run the `-v` flag, it is launch as an independent process and no output is given by the terminal

That tells me the opposite actually, it seems the problem is directly with the extension, unfortunately for me haha!

If you are willing to do a bit more hardcore debugging, please do the following:
1) run ulauncher in developer mode with extensions turned off: `ulauncher --dev --no-extensions -v`
2) go to extensions tab in ulauncher preferences, click on the Ulauncher Spotify API extension, copy the command to manually start the extension and run this command in a separate terminal. It should be something along the lines of `VERBOSE=1 ULAUNCHER_WS_API=ws://127.0.0.1:5054 ....... /ulauncher/extensions/ulauncher-spotify-api/main.py`.

This separate window should now output the proper log from the extension.

**Comment by gadc1996** [comment]
Sorry for the delay.
i used this command
```sh
VERBOSE=1 ULAUNCHER_WS_API=ws://127.0.0.1:5054/com.github.the-lay.ulauncher-spotify-api PYTHONPATH=/usr/lib/python3/dist-packages /usr/bin/python3 /home/alex/.local/share/ulauncher/extensions/com.github.the-lay.ulauncher-spotify-api/main.py 2>errors.txt 
```

Here is the output 
```sh
/usr/lib/python3/dist-packages/ulauncher/utils/display.py:2: PyGIWarning: Gdk was imported without specifying a version first. Use gi.require_version('Gdk', '3.0') before import to ensure that the right version gets loaded.
  from gi.repository import Gdk, GdkX11
[1;35mcom.github.the-lay.ulauncher-spotify-api[0m | 2021-03-18 08:31:59,730 | DEBUG | ulauncher.api.client.Client: on_message() | Incoming event PreferencesEvent
[1;35mcom.github.the-lay.ulauncher-spotify-api[0m | 2021-03-18 08:31:59,730 | DEBUG | __main__: on_preferences() | Received preferences event: {'main_keyword': 'sp', 'auth_port': '6666', 'clear_cache': 'No', 'show_help': 'Yes', 'aliases': 's: search; song: track; vol: volume; like: save; ?: help', 'search_results_limit': '8', 'request_timeout': '0.5'}
[1;35mcom.github.the-lay.ulauncher-spotify-api[0m | 2021-03-18 08:31:59,730 | DEBUG | __main__: on_preferences_update() | Received preferences update event for auth_port, changing from 8080 to 6666
[1;35mcom.github.the-lay.ulauncher-spotify-api[0m | 2021-03-18 08:31:59,730 | DEBUG | __main__: _generate_api() | Generating Spotipy object
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/ulauncher/api/client/Client.py", line 54, in on_message
    self.extension.trigger_event(event)
  File "/usr/lib/python3/dist-packages/ulauncher/api/client/Extension.py", line 52, in trigger_event
    action = listener.on_event(event, self)
  File "/home/alex/.local/share/ulauncher/extensions/com.github.the-lay.ulauncher-spotify-api/main.py", line 278, in on_event
    return self.on_preferences(event.preferences)
  File "/home/alex/.local/share/ulauncher/extensions/com.github.the-lay.ulauncher-spotify-api/main.py", line 294, in on_preferences
    self._generate_api()
  File "/home/alex/.local/share/ulauncher/extensions/com.github.the-lay.ulauncher-spotify-api/main.py", line 117, in _generate_api
    self.api = spotipy.Spotify(auth_manager=auth)
  File "/home/alex/.local/lib/python3.7/site-packages/spotipy/client.py", line 166, in __init__
    self._build_session()
  File "/home/alex/.local/lib/python3.7/site-packages/spotipy/client.py", line 200, in _build_session
    status_forcelist=self.status_forcelist)
TypeError: __init__() got an unexpected keyword argument 'allowed_methods'
[1;35mcom.github.the-lay.ulauncher-spotify-api[0m | 2021-03-18 08:32:42,157 | ERROR | ulauncher.api.client.Client: on_error() | WS Client error 
[1;35mcom.github.the-lay.ulauncher-spotify-api[0m | 2021-03-18 08:32:42,159 | WARNING | ulauncher.api.client.Client: on_close() | Connection closed. Exiting
[1;35mcom.github.the-lay.ulauncher-spotify-api[0m | 2021-03-18 08:32:42,159 | DEBUG | __main__: on_system_exit() | Received system exit event

```

**Comment by the-lay** [comment]
Thanks that is very helpful!

This seems to be Spotipy's problem: https://github.com/plamere/spotipy/issues/647

Can you please try updating spotipy (`/usr/bin/python3 -m pip install "spotipy>=2.17.1"`), it should be fixed now.

**Comment by gadc1996** [comment]
The problem is that i get this output telling me its already up to date
```sh
Requirement already satisfied: spotipy>=2.17.1 in /usr/local/lib/python3.7/dist-packages (2.17.1)
Requirement already satisfied: urllib3>=1.26.0 in /usr/local/lib/python3.7/dist-packages (from spotipy>=2.17.1) (1.26.4)
Requirement already satisfied: requests>=2.25.0 in /usr/local/lib/python3.7/dist-packages (from spotipy>=2.17.1) (2.25.1)
Requirement already satisfied: six>=1.15.0 in /usr/local/lib/python3.7/dist-packages (from spotipy>=2.17.1) (1.15.0)
Requirement already satisfied: idna<3,>=2.5 in /usr/lib/python3/dist-packages (from requests>=2.25.0->spotipy>=2.17.1) (2.6)
Requirement already satisfied: chardet<5,>=3.0.2 in /usr/lib/python3/dist-packages (from requests>=2.25.0->spotipy>=2.17.1) (3.0.4)
Requirement already satisfied: certifi>=2017.4.17 in /usr/lib/python3/dist-packages (from requests>=2.25.0->spotipy>=2.17.1) (2018.8.24)
```

**Comment by the-lay** [comment]
The plot thickens - I've updated myself and it broke for me in exactly the same way haha.

The quick fix is to roll back to the version that definitely works: `/usr/bin/python3 -m pip install "spotipy==2.16.1" "requests==2.23.0"`. Let me know if that helps.

**Comment by gadc1996** [comment]
I think i broke it even worse, now i do get an error when i got to ulauncher's settings -> extension -> Ulauncher Spotify API, i get this message:

"The extension was terminated. Please check the logs"

> go to extensions tab in ulauncher preferences, click on the Ulauncher Spotify API extension, copy the command to manually start the extension and run this command in a separate terminal. It should be something along the lines of VERBOSE=1 ULAUNCHER_WS_API=ws://127.0.0.1:5054 ....... /ulauncher/extensions/ulauncher-spotify-api/main.py.

If it is of some help, if i run this command i get the following output
```sh
/usr/lib/python3/dist-packages/ulauncher/utils/display.py:2: PyGIWarning: Gdk was imported without specifying a version first. Use gi.require_version('Gdk', '3.0') before import to ensure that the right version gets loaded.
  from gi.repository import Gdk, GdkX11
```

**Comment by the-lay** [comment]
> If it is of some help, if i run this command i get the following output
> 
> ```shell
> /usr/lib/python3/dist-packages/ulauncher/utils/display.py:2: PyGIWarning: Gdk was imported without specifying a version first. Use gi.require_version('Gdk', '3.0') before import to ensure that the right version gets loaded.
>   from gi.repository import Gdk, GdkX11
> ```

I also see the same message, but it is just a warning (coming from base ulauncher code) and can be ignored. Is this the whole output?

**Comment by the-lay** [comment]
@gadc1996 sorry to ping you, but have you tried running it again, is the issue fixed for you? I tried reinstalling it on a clean Ubuntu 20.10 today and it worked fine for me now, after the last commit where I froze the dependencies.

**Comment by the-lay** [comment]
Well, I consider that the problem is fixed now :)  
Please feel free to reopen or create a new issue one if that's not the case.

**Comment by sebastienmaurette** [comment]
@the-lay hi, fresh install for me on the last ubuntu version and i got 'loading' indefinetly

**Comment by the-lay** [comment]
Hi @sebastienmaurette, thanks for the bug report. Can you try running `/usr/bin/python3 -m pip show spotipy` and let me know which version of spotipy you have installed?

**Comment by sebastienmaurette** [comment]
Name: spotipy
Version: 2.19.0
Summary: A light weight Python library for the Spotify Web API
Home-page: https://spotipy.readthedocs.org/
Author: @plamere
Author-email: paul@echonest.com
License: MIT
Location: /home/sebastien/.local/lib/python3.8/site-packages
Requires: requests, urllib3, six
Required-by:

**Comment by sebastienmaurette** [comment]
i got this log when i try to start a request 
`com.github.the-lay.ulauncher-spotify-api | 2021-12-28 11:11:53,004 | DEBUG | ulauncher.api.client.Client: on_message() | Incoming event KeywordQueryEvent
com.github.the-lay.ulauncher-spotify-api | 2021-12-28 11:11:53,005 | DEBUG | __main__: on_event() | Extension language is: en. Searching translation files in /home/sebastien/.local/share/ulauncher/extensions/com.github.the-lay.ulauncher-spotify-api/locales.
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/ulauncher/api/client/Client.py", line 54, in on_message
    self.extension.trigger_event(event)
  File "/usr/lib/python3/dist-packages/ulauncher/api/client/Extension.py", line 52, in trigger_event
    action = listener.on_event(event, self)
  File "/home/sebastien/.local/share/ulauncher/extensions/com.github.the-lay.ulauncher-spotify-api/main.py", line 311, in on_event
    return self.on_keyword_query(event.get_keyword(), event.get_argument())
  File "/home/sebastien/.local/share/ulauncher/extensions/com.github.the-lay.ulauncher-spotify-api/main.py", line 350, in on_keyword_query
    if self.api.auth_manager.get_cached_token() is None:
AttributeError: 'NoneType' object has no attribute 'auth_manager'
`

And all my dependencies are up to date :+1: 

Name: spotipy
Version: 2.19.0

Name: urllib3
Version: 1.26.7

Name: requests
Version: 2.26.0

Maybe there it could be linked with this issue ?
https://github.com/plamere/spotipy/issues/647

**Comment by the-lay** [comment]
> Name: spotipy Version: 2.19.0 Summary: A light weight Python library for the Spotify Web API Home-page: https://spotipy.readthedocs.org/ Author: @plamere Author-email: [paul@echonest.com](mailto:paul@echonest.com) License: MIT Location: /home/sebastien/.local/lib/python3.8/site-packages Requires: requests, urllib3, six Required-by:

That's the problem - the extension requires and installs specific version of spotipy, 2.16.1. You might have updated/installed it yourself not according to the [instructions](https://github.com/the-lay/ulauncher-spotify-api#installation). Please downgrade it: `/usr/bin/python3 -m pip install "spotipy==2.16.1" "requests==2.23.0" --force-reinstall`.

If you need a newer version of spotipy for your other projects please consider using venv or anaconda for them.

**Comment by sebastienmaurette** [comment]
Thanks for your reply it's ok now.
And sorry i didn't read the instructions with attention ...

**Comment by the-lay** [comment]
No problems, glad it is solved!
